Post metrics to datadog after rotation

enicholson125 opened the pull request:
This adds code to post metrics to datadog after rotation.

It also logs errors from posting metrics in non-rotation mode. I'm on the fence about whether this is a good idea, as the current behaviour will call datadog even when the user hasn't set a metric name or authenticated, so this could make some people's implementations noisier. It still seems better to let them know what's going wrong, though.

enicholson125 commented:
This has been tested by checking that isDatadogSet returns the correct values depending on whether the datadog struct is set in the config and by checking that a key rotation with the datadog struct set results in the values being posted to datadog after the key rotation has completed.

eversC reviewed (changes requested):
Not necessarily requesting a change here, but just a question. The `postMetric()` func checks if Datadog is set using `len(apiKey)`, could we use the same thing (and perhaps use it as well in the postMetric func)?

enicholson125 commented:
Great point, I dunno how I managed to miss that! I've moved the API key length check into the datadog presence check and added a check in the non-rotate case.

eversC reviewed (approved):
lgtm 🥑
